spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/DecisionTreePredictor.java

```diff
@@ -50,112 +32,32 @@
-                                                                  config.predictionField);
-    JavaRDD<StructuredRecord> output = input.map(new Function<StructuredRecord, StructuredRecord>() {
-      @Override
-      public StructuredRecord call(StructuredRecord record) throws Exception {
-        List<Double> featureList = new ArrayList<>();
-        List<Integer> featureIndex = new ArrayList<>();
-        int counter = 0;
-        for (String field : fields.keySet()) {
-          if (record.get(field) != null) {
-            featureList.add(((Number) record.get(field)).doubleValue());
-            featureIndex.add(counter);
-          }
-          counter++;
-        }
-
-        Vector vector = Vectors.sparse(counter, Ints.toArray(featureIndex), Doubles.toArray(featureList));
-        double prediction = loadedModel.predict(vector);
-
-        return SparkUtils.cloneRecord(record, outputSchema, config.predictionField)
-          .set(config.predictionField, prediction)
-          .build();
-      }
-    });
-    return output;
+  public double predict(Vector vector) {
+    return loadedModel.predict(vector);
   }
 
   /**
    * Configuration for the DecisionTreePredictor.
    */
-  public static class DecisionTreePredictorConfig extends PluginConfig {
-
-    @Description("The name of the FileSet to load the model from.")
-    private final String fileSetName;
-
-    @Description("Path of the FileSet to load the model from.")
-    private final String path;
-
-    @Nullable
-    @Description("A comma-separated sequence of fields to be used for prediction. If both featuresToInclude and \n" +
-      "featuresToExclude are empty, all fields will be used for prediction. Features to be used, must be from one of " +
-      "the following types: int, long, float or double. Both featuresToInclude and featuresToExclude fields cannot " +
-      "be specified.")
-    private final String featuresToInclude;
-
-    @Nullable
-    @Description("A comma-separated sequence of fields to be excluded for prediction. If both featuresToInclude " +
-      "and featuresToExclude are empty, all fields will be used for prediction. Both featuresToInclude and " +
-      "featuresToExclude fields cannot be specified.")
-    private final String featuresToExclude;
-
-    @Description("The field on which to set the prediction. It will be of type double.")
-    private final String predictionField;
-
-    public DecisionTreePredictorConfig(String fileSetName, String path, @Nullable String featuresToInclude,
-                                       @Nullable String featuresToExclude, String predictionField) {
-      this.fileSetName = fileSetName;
-      this.path = path;
-      this.featuresToInclude = featuresToInclude;
-      this.featuresToExclude = featuresToExclude;
-      this.predictionField = predictionField;
-    }
+  public static class DecisionTreeConfig extends MLPredictorConfig {
```

Comment:
don't need this class if there are no additional fields and methods. Just use a MLPredictorConfig.